pennylane/transforms/op_transforms.py

```diff
@@ -457,6 +461,9 @@ def wrapper(*args, **kwargs):
                 nonlocal wire_order
                 tape, verified_wire_order = _make_tape(obj, wire_order, *args, **kwargs)
 
+                if len(getattr(tape, "operations", [])) == 1 and self._tape_fn is None:
+                    tape = tape.operations[0]
+

```

Comment:
Is this a behaviour we will want for all operation transforms that might act on tapes with a single operation? :thinking:

Reply:
Ah, I didn't initially intend for this behaviour 🙂 I was forced into it (for now) because of cases like `qml.generator(qml.adjoint(op))`. Currently, `qml.adjoint(op)` returns a _tape_ containing a single op, which is a bit unfortunate.

Reply:
So this is essentially a hotfix
```suggestion
                # HOTFIX: some operator transforms return a tape containing
                # a single transformed operator. As a result, for now we need
                # to treat a tape with a single operation as a single operation.
                if len(getattr(tape, "operations", [])) == 1 and self._tape_fn is None:
                    tape = tape.operations[0]

```

Reply:
Ah, I see! But are the side effects it might have foreseeable? And is there a plan to change the behaviour of `adjoint` to allow to remove this fix?

Reply:
Yep :) It should be as simple as rewriting `adjoint` to use `@op_transform` to remove this behaviour.